datahub-web-react/src/app/shared/handleGraphQLError.ts

```diff
@@ -0,0 +1,24 @@
+import { ErrorResponse } from '@apollo/client/link/error';
+import { message } from 'antd';
+import { ErrorCodes } from './constants';
+
+interface Props {
+    error: ErrorResponse;
+    permissionMessage: string;
+    defaultMessage: string;
+}
+
+export default function handleGraphQLError({ error, permissionMessage, defaultMessage }: Props) {
```

Comment:
Nice I really appreciate this. 

We can probably also have defaults for permission message / default message in caseyn start to add more specific error codes if we want... 

Currently we have 

400 -> Malformed query 
401 / 403 -> Auth
500 -> Server error 

but ideally we can continue to add to this!

Reply:
yes definitely! that was my thought as well - this can just handle all of the errors we care about in a more standard way - making it easier to provide specific messages for your operations.

i'll add the 400 and 500 situations as well and also handle 401 as well as 403

Reply:
so yeah we actually do handle 401 (not logged in) in `App.tsx` and there we redirect you to the login page so that should be good to go - i can create a const for not logged in in the new enum i have and use that as well